### PR TITLE
ref(tests): default to new golang deis CLI

### PR DESCRIPTION
**Pull request by mboersma**
This makes client-go/deis the default `deis` binary used in tests. 

Once it is merged, we can decide how to deprecate the old CLI and promote the new one within the codebase to make it obvious--that will be a separate PR. (See #4341.)

### CODE REVIEW

**Comment by mboersma** [comment]
I see this in the integration tests, so that's at least one thing remaining to fix:
```
limits:set --app=configsample cmd=256MB
cmd=256MB doesn't fit format type=#unit or type=#
Examples: web=2G worker=500M web=300
Applying limits... done
```
It should have been (from the old python client):
```
limits:set --app=configsample cmd=256MB
Applying limits... done, v10
```

**Comment by Joshua-Anderson** [comment]
@mboersma That's due to the fact, I thought we only supported `B`, `M`, or `G`, for memory units. Is there a complete list of all the memory units we support?

**Comment by Joshua-Anderson** [comment]
The tests failed with:

```
Error: Get http://deis.test-46b84a697e.deis.works/v1/apps/healthchecksample/containers/: EOF
```

For some reason the go client frequently throws this error when running `ps:list`.

**Comment by bacongobbler** [comment]
perhaps that's the equivalent of the python client's `BadStatusLine` error we keep seeing in the tests? Not sure...

File: pkg/prettyprint/colorizer_test.go [hunk 1]
```diff
@@ -67,16 +68,13 @@ func TestPrettyTabs(t *testing.T) {
...
-`
-	otherExpected := `foo  bar
+	expected := `foo  bar
 test testing
 `
 
 	output := PrettyTabs(test, 1)
 
-	if !(output == expected || output == otherExpected) {
+	if !(output == expected) {
```
Comment (carmstrong) [inline pkg/prettyprint/colorizer_test.go]: Wouldn't this read better as `if output != expected`?

**Comment by carmstrong** [comment]
Code LGTM.

**Comment by mboersma** [comment]
I've now seen two EOF errors during test_ps.go, although I have tried and failed to reproduce this using the same commands interactively. So there is still something to fix here... I can try `DisableKeepalives=true` but it's difficult to know if that fixes things unless I can reproduce it locally.

**Comment by krancour** [comment]
Related to #4158... which I just closed, but we can re-open that if this warrants it.

**Comment by mboersma** [comment]
So several of us have been testing this, as well as Jenkins. EOF errors are rare but still there. We think this is a different manifestation of the `BadStatusLine` error that cropped up sometimes using the old python CLI. @bacongobbler got some logs from nginx with debug on that seem to point to this: http://serverfault.com/questions/716862/recv-not-ready-11-resource-temporarily-unavailable-nginx-1-1-19-supervisor

tl:dr, still trying to pin down the EOF, but its impact appears to be similar to a problem we already had, so this is probably ready to :ship:.

**Comment by bacongobbler** [comment]
is this superseded by #4341 or is this different?

**Comment by mboersma** [comment]
> is this superseded by #4341 or is this different?

I intended them to be separate in order to facilitate reviewing, but at this point it's probably simpler to close this and focus on #4341.
